Sources/Subs-Calendar.php

```diff
@@ -1025,6 +1143,302 @@ function getEventProperties($event_id)
 }
 
 /**
+ * Gets an initial set of date and time values for creating a new event.
+ *
+ * @return array An array containing an initial set of date and time values for an event.
+ */
+function getNewEventDatetimes()
+{
+	global $user_info;
+
+	$today = getdate();
+
+	$tz = isset($_REQUEST['tz']) ? $_REQUEST['tz'] : getUserTimezone();
+	$allday = isset($_REQUEST['allday']) ? $_REQUEST['allday'] : (isset($_REQUEST['start_time']) ? 0 : 1);
+	$span = isset($_REQUEST['span']) ? $_REQUEST['span'] : 1;
+	$start_year = isset($_REQUEST['year']) ? $_REQUEST['year'] : $today['year'];
+	$start_month = isset($_REQUEST['month']) ? $_REQUEST['month'] : $today['mon'];
+	$start_day = isset($_REQUEST['day']) ? $_REQUEST['day'] : $today['mday'];
+	$start_hour = isset($_REQUEST['hour']) ? $_REQUEST['hour'] : $today['hours'];
+	$start_minute = isset($_REQUEST['minute']) ? $_REQUEST['minute'] : $today['minutes'];
+	$start_second = isset($_REQUEST['second']) ? $_REQUEST['second'] : $today['seconds'];
+	$end_year = isset($_REQUEST['end_year']) ? $_REQUEST['end_year'] : $today['year'];
+	$end_month = isset($_REQUEST['end_month']) ? $_REQUEST['end_month'] : $today['mon'];
+	$end_day = isset($_REQUEST['end_day']) ? $_REQUEST['end_day'] : $today['mday'];
+	$end_hour = isset($_REQUEST['end_hour']) ? $_REQUEST['end_hour'] : ($today['hours'] < 23 ? $today['hours'] + 1 : $today['hours']);
+	$end_minute = isset($_REQUEST['end_minute']) ? $_REQUEST['end_minute'] : ($today['hours'] < 23 ? $today['minutes'] : 59);
+	$end_second = isset($_REQUEST['end_second']) ? $_REQUEST['end_second'] : 0;
+	$start_date = isset($_REQUEST['start_date']) ? $_REQUEST['start_date'] : sprintf('%04d-%02d-%02d', $start_year, $start_month, $start_day);
+	$start_time = isset($_REQUEST['start_time']) ? $_REQUEST['start_time'] : sprintf('%02d:%02d:%02d', $start_hour, $start_minute, $start_second);
+	$end_date = isset($_REQUEST['end_date']) ? $_REQUEST['end_date'] : sprintf('%04d-%02d-%02d', $end_year, $end_month, $end_day);
+	$end_time = isset($_REQUEST['end_time']) ? $_REQUEST['end_time'] : sprintf('%02d:%02d:%02d', $end_hour, $end_minute, $end_second);
+	$start_datetime = isset($_REQUEST['start_datetime']) ? $_REQUEST['start_datetime'] : $start_date . ' ' . $start_time;
+	$end_datetime = isset($_REQUEST['end_datetime']) ? $_REQUEST['end_datetime'] : $end_date . ' ' . $end_time;
```

Comment:
Good catch. Looks like I forgot to finish implementing the validation in this function. I've now added that in dfc0e8e.